### ISSUE TITLE
Feature Request: Support buildx for users with M1 ARM chips

### ISSUE THREAD

**Issue by ghost** [cuber-cloud/cuber-gem#2]
Hello,

Just tried this gem using the example Docker file and it failed on Digital Ocean because the docker image was built on my machine which is a M1 Mac.

Is it possible to support docker buildx and the cuber docs updated with example of how to use it?

Thank you for a great Gem that could save me ages. It was very easy until this issue.

This might be a feature I could add, since I'm stuck on it.

Maybe in the Cuberfile it has option.

buildx 'linux/amd64'

**Comment by collimarco** [comment]
I have found this:
https://dev.to/lakhansamani/create-docker-image-on-apple-silicon-m1-mac-2f75

Probably we just need to pass a `--platform` option to `docker build` ([here](https://github.com/cuber-cloud/cuber-gem/blob/master/lib/cuber/commands/deploy.rb#L67)).

**Comment by ghost** [comment]
Thank you for quick response. That worked, will this option be added to the project to enable it in the Cuberfile? 

Also, after editing the gem and running it with that option my next error was to do the migration, it tried to run in development with the example provided.

Changing to this worked.

```
migrate 'env RAILS_ENV=production rails db:migrate', check: 'env RAILS_ENV=production rake db:abort_if_pending_migrations'
```

Now, it looks like I just need to setup some ENV for postgres username and password and it might work.

Best project since Capistrano, it's really going to make deploying much easier, Thank you again.

**Comment by ghost** [comment]
Can Cuber be used to setup Postgres in Kubernetes also? or is it better/ recommended to setup on it's own server and just point to it from the database.yml ?

How do you recommend doing that on Digital Ocean?

Just had a look and Digital Ocean has a databases add on, so I'm guessing that is the way forward, sorry for the question, I'll stop spamming your project now.

**Comment by collimarco** [comment]
> That worked, will this option be added to the project to enable it in the Cuberfile?

Yes, I should add it to the next release of Cuber. In any case I think that adding `--platform linux/amd64` should work on all major platforms (DigitalOcean, AWS, Google, etc.) so it can be hardcoded, without adding an unnecessary option to the Cuberfile.

> migrate 'env RAILS_ENV=production rails db:migrate'

You should define `RAILS_ENV` using the Cuber `env` command (in the Cuberfile). That is the correct approach.

> Can Cuber be used to setup Postgres in Kubernetes also? or is it better/ recommended to setup on it's own server and just point to it from the database.yml ?

Currently we don't run Postgresql on Kubernetes - there are too many pitfalls in my opinion. I prefer to use a VM (Droplet) on DigitalOcean (or alternatively their managed service). A droplet is about 5 times cheaper compared to their managed database (with same hardware), but you need some more skills and maintenance.

**Comment by ghost** [comment]
Thank you, I'm all good now, closing ticket.

**Comment by collimarco** [comment]
@leigh-mackay I have just released v1.2.1 of this gem which fixes this issue.
